File: python/day10.py
import itertools

def look_and_say(seq):
    return "".join([str(len(list(g))) + k for k, g in itertools.groupby(seq)])

input = "1113222113"

for i in range(50):
    input = look_and_say(input)
    if i == 39:
        print(len(input))
print(len(input))

File: aoc2015/src/bin/day10.rs
extern crate itertools;

use itertools::Itertools;

fn look_and_say(seq: &str) -> String {
    seq.chars()
        .into_iter()
        .group_by(|a| a.clone())
        .map(|i| i.1.len().to_string() + &i.0.to_string())
        .fold("".to_string(), |a, b| a + &b)
}

fn main() {
    let mut state = "1113222113".to_string();

    for _ in 0..40 {
        state = look_and_say(&state);
    }
    println!("{}", state.len());

    for _ in 0..10 {
        state = look_and_say(&state);
    }
    println!("{}", state.len());
}
